cumulusci/tasks/preflight/recordtypes.py

```diff
@@ -0,0 +1,21 @@
+from cumulusci.tasks.salesforce import BaseSalesforceApiTask
+
+class GetRecordTypesForSobject(BaseSalesforceApiTask):
+    task_options = {
+        "sobject": {
+            "description": "The sObject for which we want check if any recordtypes exist",
+            "required": True,
+        }
+    }
+
+    def _run_task(self):
+        sobject = self.options["sobject"]
+        results = self.tooling.query(
+            f"Select  Name FROM RecordType WHERE SobjectType = '{sobject}'"
```

Comment:
Question: I know that many Tooling API queries cannot be bulkified. Does this one successfully fetch, say, all Record Types? If it does, we could potentially increase performance and flexibility of this task by digesting all Record Types into a dict, and then let the `when` expression specify the object rather than a task option (e.g., `when: "Opportunity" not in tasks.check_sobject_record_types()`).

```suggestion
            f"Select Name FROM RecordType WHERE SobjectType = '{sobject}'"
```

Reply:
This is a great idea and my original thought - but wasnt sure how we measure expense. This query will fetch all existing RecordTypes. From what I could tell that probably wont ever be a 'huge' number, but at the same time I dont see many hard limits to how many recordtypes per object one could have. 
Will make this suggested change!

Reply:
@ClayTomerlin  I think the pattern you can best imitate, if you'd like a pattern, is in the CheckSobjectsAvailable task, where `return_values` is a dict. You can make the keys sObject names and the values a list of Record Types.

Reply:
@davidmreed 
That makes sense, is there any way to return more than the sObjectType names (basically more than the one field in the query)? This would allow expansion to also confirm a 'specific' recordtype exists. 
For now Ill make it simple

Reply:
Yeah, the route noted above can do that. If the return value has a structure like this:

`{ 'Opportunity': ['MyRTDeveloperName']}`

you can write `when` clauses like

`when: 'Account' not in tasks.check_sobject_record_types()`

and

`when: 'Opportunity' in tasks.check_sobject_record_types() and 'MyRTDeveloperName' in tasks.check_sobject_record_types()['Opportunity']`

So you cover both bases - checking for _any_ record type, and checking for a _specific_ record type.